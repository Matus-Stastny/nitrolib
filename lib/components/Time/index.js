"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _format = _interopRequireDefault(require("date-fns/format"));

var _context = require("../../services/intl/context");

var _LangInfo = require("../../records/LangInfo");

var Time = function Time(_ref) {
  var time = _ref.time;
  return React.createElement(_context.Consumer, null, function (intl) {
    return (0, _format.default)(time, (0, _LangInfo.fixTimeFormat)(intl.language.timeFormat));
  });
};

var _default = Time;
exports.default = _default;