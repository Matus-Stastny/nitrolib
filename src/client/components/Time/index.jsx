// @flow strict
import * as React from "react";
import format from "date-fns/format"; /* flowlint-line nonstrict-import:off */

import { Consumer } from "client/services/intl/context";

type Props = {
  time: Date,
};

const Time = (props: Props) => (
  <Consumer>{intl => format(props.time, intl.language.timeFormat)}</Consumer>
);

export default Time;
