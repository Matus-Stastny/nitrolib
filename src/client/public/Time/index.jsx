/* @flow */
import withIntl from "../services/intl/withIntl";
import { DEFAULT_TIME } from "../services/intl/consts";

type Props = {
  time: Date,
  locale: string,
  format?: { [key: string]: number | string },
};

const Time = (props: Props) =>
  new Intl.DateTimeFormat(props.locale, props.format).format(props.time);

Time.defaultProps = {
  format: DEFAULT_TIME,
};

export default withIntl(Time);
