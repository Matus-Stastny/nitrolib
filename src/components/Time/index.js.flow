// @flow strict
import * as React from "react";

type Props = {|
  time: Date,
|};

declare export default React.ComponentType<Props>;
