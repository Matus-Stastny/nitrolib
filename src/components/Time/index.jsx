// @flow strict
import * as React from "react";
import format from "date-fns/format";

import { Consumer } from "../../services/intl/context";

type Props = {
  time: Date,
};

const Time = (props: Props) => (
  <Consumer>{intl => format(props.time, intl.language.timeFormat)}</Consumer>
);

export default Time;
