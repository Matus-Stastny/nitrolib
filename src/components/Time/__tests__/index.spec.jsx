// @flow strict
import * as React from "react";
import { mount } from "enzyme";

import Time from "..";

describe("#Time", () => {
  test("format", () => {
    const time = new Date(Date.UTC(2019, 2, 8, 11, 30));
    const wrapper = mount(<Time time={time} />);

    console.log(time);

    expect(wrapper.contains("12:30")).toBe(true);
  });
});
