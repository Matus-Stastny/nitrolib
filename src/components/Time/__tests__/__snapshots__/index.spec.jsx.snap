// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Time format 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
